src/Contacts/src/CNLabeledValue.m

```diff
@@ -22,35 +22,69 @@
 @implementation CNLabeledValue
 
 - (instancetype)initWithLabel:(NSString *)label value:(id)value {
+    if (self = [super init]) {
+//        [self applyLabeledValueVariablesIdentifier: [[NSUUID alloc] init].UUIDString label:label value:value];
+        [self applyLabeledValueVariablesIdentifier: [[NSUUID alloc] init].UUIDString label:label value:value];
+    }
+    
+    return self;
+}
+
+- (instancetype)initWithIdentifier:(NSString *)identifier label:(NSString *)label value:(id)value {
+    if (self = [super init]) {
+        [self applyLabeledValueVariablesIdentifier:identifier label:label value:value];
+    }
+    
+    return self;
+}
+
+- (void)dealloc {
+    [_identifier release];
+    [_label release];
+    [_value release];
+    [super dealloc];
+}
+
+- (instancetype)applyLabeledValueVariablesIdentifier:(NSString *)identifier label:(NSString *)label value:(id)value {
+    if (value == nil) {
+        @throw [[NSException alloc] initWithName:@"CNPropertyInvalidTypeExpression" reason:@"Variable value cannot be null." userInfo:nil];
+    }
+    
+    _identifier = [identifier copy];
+    _label = [label copy];
+    _value = [value copy];
     return self;
 }
 
 + (instancetype)labeledValueWithLabel:(NSString *)label value:(id)value {
-    id new_class = [[[self class] alloc] init];
-    return new_class;
+    return [[[[self class] alloc] initWithLabel:label value:value] autorelease];
 }
 
 - (instancetype)labeledValueBySettingLabel:(NSString *)label {
-    return self;
+    return [[[[self class] alloc] initWithIdentifier:_identifier label:label value:_value] autorelease];
 }
 
 - (instancetype)labeledValueBySettingLabel:(NSString *)label value:(id)value {
-    return self;
+    return [[[[self class] alloc] initWithIdentifier:_identifier label:label value:value] autorelease];
 }
 
 - (instancetype)labeledValueBySettingValue:(id)value {
-    return self;
+    return [[[[self class] alloc] initWithIdentifier:_identifier label:_label value:value] autorelease];
 }
 
+/*
+ TODO: Add support for obtaining localized string from contact label
+ */
 + (NSString *)localizedStringForLabel:(NSString *)label {
-    return @"";
+    // If you provide a custom label, it will return the same label.
+    return label;
 }
 
 
 // NSCopy
-- (nonnull id)copyWithZone:(nullable NSZone *)zone {
+- (id)copyWithZone:(NSZone *)zone {
     id copy = [[[self class] alloc] init];
```

Comment:
This will create a new empty instance, not a copy of `self`.

Also note that you cannot really call plain `init` if your class has custom inits (and does not override the plain `init`). It will compile, but it will fail at runtime much later when you're expecting something to be initialized and it's not.

Reply:
Yeah... I didn't really focusing on implementing NSCopying or NSSecureCoding. I wanted to focused on the main functions first before I implement these properly.

I will definitely keep your comments in mind when I actually implement these properties.